keras/engine/topology.py

```diff
@@ -465,6 +465,17 @@ def __call__(self, x, mask=None):
             x: can be a tensor or list/tuple of tensors.
             mask: tensor or list/tuple of tensors.
         '''
+        # place the op on the right device:
+        if type(x) in [list, tuple] and not hasattr(x[0], '_keras_on_device'):
```

Comment:
Theano doesn't let you explicitly assign an op to device. Rather a device is chosen implicitly by looking at the inputs to the op.